DataFormats/GeometrySurface/interface/SOARotation.h

```diff
@@ -12,7 +12,7 @@ class TkRotation;
 template <class T>
 class SOARotation {
 public:
-  constexpr inline SOARotation() {}
+  inline SOARotation() {}
```

Comment:
Why is `constexpr` removed?

Reply:
It all starts with [`ParamsOnGPUT`](https://cmssdt.cern.ch/dxr/CMSSW/source/RecoLocalTracker/SiPixelRecHits/interface/pixelCPEforGPU.h#84). This `struct` contains device memory pointers, which are then used as parameters for kernels. 
This was now made [to contain the instances of the `structs` themselves](https://github.com/cms-sw/cmssw/blob/58b4bbadb36878db3ebdfe04f558c4f2d63cfc15/DataFormats/PixelCPEFastParams/interface/PixelCPEFastParams.h#L112-L130), so that we can wrap the [`ParamsOnDeviceT`](https://github.com/cms-sw/cmssw/blob/58b4bbadb36878db3ebdfe04f558c4f2d63cfc15/DataFormats/PixelCPEFastParams/interface/PixelCPEFastParams.h#L112-L130) (renamed from `ParamsOnGPUT`) in a `device_host` or `host_buffer` and pass it to Alpaka modules. 

However, this changes the way that `ParamsOnGPUT` is instantiated, meaning that its constructor will implicitly instantiate all `struct`s contained in `ParamsOnGPUT`. One of them is `SOAFrame`, which contains `SOARotation`. Their `constrexpr` constructors break `ParamsOnDeviceT` instantiation for some reason, which expects that the attributes of `SOAFrame` and `SOARotation` are initialized at construction time, them being considered `const`, even if they're (explicitly) not.

----------------

TL;DR: See Andrea's comment above.

Reply:
FWIW, we can probably put it back to `constexpr` once we hit C++23...